cypress/integration/buy-shirt.spec.ts

```diff
@@ -1,36 +1,77 @@
+const productsListPage = new ProductsListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    // Step 1. Go to http://automationpractice.com/index.php
-    cy.visit("http://automationpractice.com/");
-    // Step 2. Go to T-Shirt
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    // Step 3. Add to Cart first T-Shirt
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    // Step 4. Click on "Proceed to checkout" button on pop-up window
-    // (The one with the "Product successfully added..." message)
-    cy.get("[style*=\"display: block;\"] .button-container > a").click();
+    // cy.visit("http://automationpractice.com/");
+    // cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
+    // cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
+    // cy.get("[style*=\"display: block;\"] .button-container > a").click();
+    // cy.get(".cart_navigation span").click();
+    // cy.get("#email").type("aperdomobo@gmail.com");
+    // cy.get("#passwd").type("WorkshopProtractor");
+    // cy.get("button#SubmitLogin > span").click();
+    // cy.get("div#center_column button[type=\"submit\"] > span").click();
+    // cy.get("#cgv").click();
+    // cy.get("form#form button[type=\"submit\"] > span").click();
+    // cy.get("div#HOOK_PAYMENT div:nth-child(1) > div > p > a").click();
+    // cy.get("p#cart_navigation span").click();
+    // cy.get("#center_column > div > p > strong").should("have.text", "Your order on My Store is complete.");
+
+    // // Step 1. Go to http://automationpractice.com/index.php
+    menuContentPage.visitMenuContentPage();
+    // // Step 2. Go to T-Shirt
+    menuContentPage.goToTShirtMenu();
+    // // Step 3. Add to Cart first T-Shirt
+    productsListPage.addProductToCart();
+    // // Step 4. Click on "Proceed to checkout" button on pop-up window
+    // // (The one with the "Product successfully added..." message)
+    productsListPage.continueToCheckoutPopup();
     // Step 5. Click on "Proceed to checkout" on the Summary step
-    cy.get(".cart_navigation span").click();
+    shoppingCartPage.continueToCheckoutCart();
     // Step 6. Write "Email" in "Sign In" step
-    cy.get("#email").type("aperdomobo@gmail.com");
+    loginPage.enterEmailInLogin();
     // Step 7. Write "Password" in "Sign In" step
-    cy.get("#passwd").type("WorkshopProtractor");
-    // Finish missing steps from 8 to 13...
+    loginPage.enterPasswordInLogin();
+
+    // Start creating missing steps from 8 to 13 ...
+
     // Step 8. Click "Sign In" button
-    cy.get("button#SubmitLogin > span").click();
+    loginPage.continueToCheckoutBtnSignIn();
```

Comment:
for the login page, you can have one single method, called login, and that method can receive the email and password, in that way, the login page will not be attached to one single user, it will be able to receive and login any user.  Another alternative is divide the process in two, one method to digit the credentials and another for click in the login. I prefer the first approach. Choose whatever you want :)

Reply:
Refactored